includes/service-providers/class-newspack-newsletters-service-provider.php

```diff
@@ -676,7 +670,25 @@ protected function update_contact_local_lists( $email, $lists = [], $action = 'a
 					$list = Subscription_List::from_public_id( $list_id );
 
 					if ( ! $list->is_configured_for_provider( $this->service ) ) {
-						return new WP_Error( 'List not properly configured for the provider' );
+						do_action(
+							'newspack_log',
+							'newspack_esp_update_contact_lists_error',
+							sprintf(
+								'Error handling local list: %s',
```

Comment:
NIT: We're still learning how to use the logs, so this is just a suggestion, not sure what's the best way.

But it seems that the "message" here is a bit redundant. Since it's the same in all cases, it has the same purpose of the `error_code` up there (`newspack_esp_update_contact_lists_error`)

Maybe the string that is down there as `errors` in the `data` array could be added here and removed from there.

Also. Not sure if it's a good thing to add this dynamic value of the ID here in the message. This will give us an infinite number of possible values making it a little more difficult to filter by one specific error. Maybe the ID can go down there in the data?

Reply:
@leogermani makes sense. 710ea24 updates the error logs so that the messages are evergreen in both cases, and moves the `$list_id` to a data param instead.